### PR TITLE
Added TimeStamp to profile.

### LINKED ISSUE #2: Add timestamp to the profile
**Describe the feature you'd like**
Add the time of creation below the profile name as shown in the below image.

**Screenshots**

**Would you like to work on the issue?**

- [ ] Yes
- [ ] No
- Other: open for all

### PR DESCRIPTION
**Pull request by Haid-Faiz**
Fixes #2 

Changes: Added TimeStamp feature to profile

Screenshots for the change:
Screenshot_20201010-153304_Quiet Hours

### CODE REVIEW

**Comment by Haid-Faiz** [comment]
@rob729  Now, this time only one commit is there.. Please have a look.

File: app/src/main/java/com/rob729/quiethours/Fragments/NewProfileFragment.kt [hunk 1]
```diff
@@ -127,14 +128,18 @@ class NewProfileFragment : Fragment() {
                 viewSnackBar(it, "Please enter a valid time.(Within 12 hour limit)")
             } else {
                 val daySelected = Gson()
+                // Generating Formated Time
+                var myFormatedTime = SimpleDateFormat("EEE, d MMM yyyy hh:mm").format(Date())
```
Comment (rob729) [inline app/src/main/java/com/rob729/quiethours/Fragments/NewProfileFragment.kt]: The variable is not appropriate.
Change it to currentTime.

File: app/src/main/java/com/rob729/quiethours/Fragments/NewProfileFragment.kt [hunk 2]
```diff
@@ -127,14 +128,18 @@ class NewProfileFragment : Fragment() {
...
                 val profile = Profile(
                     name = binding.userToDoEditText.text.toString(),
                     shr = shr,
                     smin = smin,
                     ehr = ehr,
                     emin = emin,
                     d = daySelected.toJson(days),
-                    colorIndex = Random.nextInt(0, 8)
+                    colorIndex = Random.nextInt(0, 8),
+                    // Passing Formatted Timestamp
```
Comment (rob729) [inline app/src/main/java/com/rob729/quiethours/Fragments/NewProfileFragment.kt]: You can remove this comment.

File: app/src/main/res/layout/item_row.xml [hunk 3]
```diff
@@ -41,6 +41,16 @@
...
                     android:transitionName="profileName" />
 
+            <TextView
+                    android:id="@+id/timeStamp"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:layout_below="@+id/ProfileName"
+                    android:layout_alignStart="@+id/ProfileName"
+                    android:layout_marginTop="2dp"
+                    android:text="TimeStamp"
```
Comment (rob729) [inline app/src/main/res/layout/item_row.xml]: Let's not pass a default value of timeStamp textview.
Either change it to tools:text="TimeStamp" or remove this line.

Comment (Haid-Faiz) [reply app/src/main/res/layout/item_row.xml]: @rob729 okk

**Review by Haid-Faiz — COMMENTED** [review]
I updated item_row.xml

**Comment by rob729** [comment]
@Haid-Faiz I have also commented on some other things also. Please also have a look at them.

**Comment by Haid-Faiz** [comment]
@rob729  i directly updated item_row.xml from github itself but now again it is showing 2 commits on github,,. So, can we squash commits directly from github....??

**Comment by Haid-Faiz** [comment]
> @Haid-Faiz I have also commented on some other things also. Please also have a look at them.

@rob729 ok

**Comment by rob729** [comment]
I don't think we can squash commits from Github.

**Comment by rob729** [comment]
You will have to learn about squashing commits. You can't escape it at least while contributing to this project.

**Review by Haid-Faiz — COMMENTED** [review]
@rob729  Updated denoted files...

**Comment by Haid-Faiz** [comment]
> You will have to learn about squashing commits. You can't escape it at least while contributing to this project.

@rob729 Ok, let me squash them from git..

**Review by rob729 — COMMENTED** [review]
@Haid-Faiz One more thing, add android:layout_marginStart="18dp" to timeStamp also as the start of profile name and timeStamp is not aligned as of now.
And make topMargin of the profile name to 6dp instead of 12dp.

**Comment by rob729** [comment]
@Haid-Faiz Share the git commands you are using for squashing the commits. It feels like this PR is also heading towards a mess.

**Comment by Haid-Faiz** [comment]
> @Haid-Faiz Share the git commands you are using for squashing the commits. It feels like this PR is also heading towards a mess.

@rob729 i am facing some problem...  I successfully squashed commits from android studio terminal but as i pushed those changes of squashed commits then it shows me error in pushing with option merge & rebase & cancel...  i also read on google that firstly we need to pull before pushing so i also did in this way....   but after successfully squashing it shows error in pushing....

- These are the commands i am using...
- git rebase -i HEAD~5
- then it shows editing list of commits.... then after presing insert button it allows to edit the file... 
- after editing, i pressed :wq to save the file then it shows commits history...
- then i removed all commits except one main commit
- then again i saved it by pressing :wq & it rebased the project successfully & squashed the commits.
- then i use git log to see commits....  then it doesn't show me squashed commits.. 

In this way i am squashing the commits..

**Comment by Haid-Faiz** [comment]
@rob729  as i am pushing after squashing then it is showing me this dialog.....
Screenshot (171)

**Comment by rob729** [comment]
I don't think you are squashing the commits properly, Have you seen the video shared in the Gitter channel properly?

**Comment by Haid-Faiz** [comment]
> I don't think you are squashing the commits properly, Have you seen the video shared in the Gitter channel properly?

**Comment by Haid-Faiz** [comment]
> I don't think you are squashing the commits properly, Have you seen the video shared in the Gitter channel properly?

@rob729 Ya i watched it properly infect i watched some more tutorial and read articles...

and when does android studio/Git shows this dialog message (which i enclosed in screenshot.)

**Comment by rob729** [comment]
git rebase -i HEAD~n where n is the number of commits you want to squash.
Then you will be shown a list of commits and then pick only one commit and select s for other commits.
Then you save the changes and after saving the changes the commit message list is shown. You keep only the relevant commit message and remove all other messages. Then save those changes and exit.
After this, you will see that instead of n commits only one commit is present.
Now when you will try to push those changes the terminal will show an error because the head of the local branch and branch present on Github is different.
Use git push -f  to push the new changes.
In this whole process, no new commit will be added.
If you follow the above instructions and the video shared in the Gitter channel then you should be able to do it properly.
Before doing this process, get rid of those unnecessary commits you have added using the git reset command.

**Comment by Haid-Faiz** [comment]
@rob729 Updated... Now only one commit is there.... Thanks for your guidance... before i wasn't doing Forced pushed... Means that's why it was causing me error. :)

**Review by rob729 — CHANGES_REQUESTED** [review]
@Haid-Faiz  change the top margin of ProfileName in item_row.xml to 6dp.
Rest looks go to me. Before committing the changes once upload the screenshot after making the changes so that I can verify if it is looking fine or not.

**Comment by Haid-Faiz** [comment]
@rob729 okk.... i forgot to update that margin

**Comment by Haid-Faiz** [comment]
@rob729  Kindly also Check my RoomDatabase migration policy code.... because i just Checked to install previous version from play store then on installing this latest version with migration policy... it's still getting crashed... Is i wrote the wrong code...??

**Comment by rob729** [comment]
@Haid-Faiz https://github.com/rob729/Quiet_Hours/suites/1324886160/artifacts/21157792 Use this link to install the app. After installing the app, create some profiles in it. Now install the app from the android studio and it should act as an update to the already installed app. Check if the app is crashing or not.
Incase the android studio prompts to uninstall the existing version of the app then inform me.

**Comment by Haid-Faiz** [comment]
@rob729 this app is not getting installed from above link...

**Comment by rob729** [comment]
Do you already have an installed version of quiet hours?

**Comment by Haid-Faiz** [comment]
@rob729 No, i deleted that before installing

**Comment by rob729** [comment]
@Haid-Faiz I have checked the code for migration. It is fine. Change the top margin of the profile name and upload the screenshot.

**Comment by Haid-Faiz** [comment]
@rob729 Okk then... Thanks

**Comment by Haid-Faiz** [comment]
@rob729  Fix #29 
Updated ProfileName top margin...

Screenshot_20201012-195915_Quiet_Hours 1

**Review by rob729 — APPROVED** [review]
@Haid-Faiz Congratulations on your 1st contribution to Quiet Hours.

**Comment by rob729** [comment]
give @Haid-Faiz  75 points

**Comment by Haid-Faiz** [comment]
> @Haid-Faiz Congratulations on your 1st contribution to Quiet Hours.

@rob729 Thanks Robin Bhaiya... Actually before SLoP i didn't work on Git & Github very much... i only used to do basic git operations.... But now my level of Git has enhanced after this project & learned about new topics like conflicts & Squashing, etc. Thanks to you & SLoP 👍 .

**Pull request merged**